### DISCUSSION TITLE
Adding font-weights

### DISCUSSION THREAD

**Discussion by jonasbirkelof** [squidfunk/mkdocs-material#7747] [Q&A]
By default it looks like the Roboto font come with the 500, 700 and 900 weights. How do I add more weights? I have tried to override the font block paritial but htat doesn't work. It changens the `<head>` section but setting `font-weight: 100` does nothing.

```
{% extends "base.html" %}

{% block fonts %}
  <link rel="preconnect" href="https://fonts.googleapis.com">
  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
  <link href="https://fonts.googleapis.com/css2?family=Roboto:wght@100;300;400;500;700;900&display=swap" rel="stylesheet">
{% endblock %}
```

**Comment by kamilkrzyskow** [comment]
Hello @jonasbirkelof,
in what way exactly does the block override not work?

It seems you've also removed a bit too much of the fonts block:

https://github.com/squidfunk/mkdocs-material/blob/b88d1f0eb9ef97c37d7eebabd131a415285249a8/src/templates/base.html#L139-L144

Have you confirmed that the woff2 font files do load in the browser tools networking tab?

**Reply by jonasbirkelof** [reply]
Yea, I skipped the mono-type just for testing.

The way it doesn't work is that the weight does not apply. I thought that if I added the weights in the link tag, I would be able to use them in my css like I usually would, But 100, 300, 400 and 500 for instance all looks the same.

**Reply by kamilkrzyskow** [reply]
I didn't mean the mono-type. I mean the `<style>` tag with the `--md-text-font` variable.
Your code as it's currently doesn't set the font-family to Roboto, only a `_` because of the missing variable.
![image](https://github.com/user-attachments/assets/90ef667a-cecf-4198-b0f4-f5f7442c045e)

---

You need to add the variables:

```html
<style>
  :root {
    --md-text-font: "Roboto";
    --md-code-font: "Roboto";
  }
</style>
```

![image](https://github.com/user-attachments/assets/53515078-1e96-4289-8303-5ceb820dac25)

Also to make it more dynamic and allow other fonts, do not skip the logic from the current implementation, just make adjustments to change the weights, the rest should be as-is:
https://github.com/squidfunk/mkdocs-material/blob/d46834475140223d2c322e5deab98e26789e04d9/src/templates/base.html#L127-L146
